tests/e2e/cucumber/steps/ui/resources.ts

```diff
@@ -281,37 +287,43 @@ export const processDownload = async (
     parentFolder = folder !== 'undefined' ? folder : null
     downloads = await pageObject.download({
       folder: parentFolder,
-      names: files,
+      resources: files,
       via: actionType === 'batch action' ? 'BATCH_ACTION' : 'SIDEBAR_PANEL'
     })
+
+    downloads.forEach((download) => {
+      downloadedResources.push(download.suggestedFilename())
+    })
+
     if (actionType === 'sidebar panel') {
       expect(files.length).toBe(downloads.length)
-      downloads.forEach((download) => {
-        expect(files).toContain(download.suggestedFilename())
-      })
+      for (const resource of files) {
+        if (resource.type === 'file') {
+          expect(downloadedResources).toContain(resource.name)
+        } else {
+          // downloading folders in oc10 downloads with name of resource but in ocis it is downloaded as 'download.tar'
+          config.ocis
+            ? expect(downloadedResources).toContain('download.tar')
```

Comment:
There are some steps in the scenarios that download multiple folders. I suppose that `downloadedResources` will contain `download.tar` and also either:
`download.tar` will be overwritten multiple times, as each folder download happens, or
there will be `download(1).tar` `download(2).tar` (with some file naming convention to avoid duplicates

Is there something we can do in the test code to get confidence that all the required folders did get downloaded?

Reply:
@phil-davis the downloaded files actually do not gets duplicated as `download(1).tar` or `download(2).tar`. it will be added to array `downloadedResources`. We have checked a step above with `expect(files.length).toBe(downloads.length)` to make sure that the downloaded array contains the no of files we have pass in data tables.